tests/unit/test_diffsync_model_flags.py

```diff
@@ -0,0 +1,196 @@
+    backend_a.get(backend_a.site, "nyc").remove_child(missing_device)
+    backend_a.remove(missing_device)
+
+    with pytest.raises(NotImplementedError):
+        exception_backend_a.sync_from(backend_a)
+
+
+def test_diffsync_sync_with_crud_no_delete_flag_on_models_skip_children(backend_a, exception_backend_a):
+    missing_device = backend_a.get(backend_a.device, "nyc-spine1")
+    backend_a.get(backend_a.site, "nyc").remove_child(missing_device)
+    backend_a.remove(missing_device)
+
+    assert "nyc-spine1" in exception_backend_a.get(exception_backend_a.site, "nyc").devices
+    assert "nyc-spine1__eth0" in exception_backend_a.get(exception_backend_a.device, "nyc-spine1").interfaces
+    assert exception_backend_a.get(exception_backend_a.interface, "nyc-spine1__eth0")
+
+    model1 = exception_backend_a.get(exception_backend_a.device, "nyc-spine1")
+    model1.model_flags |= DiffSyncModelFlags.CRUD_NO_DELETE
+    model1.model_flags |= DiffSyncModelFlags.SKIP_CHILDREN_ON_DELETE
+
+    diff = exception_backend_a.diff_from(backend_a)
+    assert diff.has_diffs()
+
+    exception_backend_a.sync_from(backend_a)
+    assert "nyc-spine1" in exception_backend_a.get(exception_backend_a.site, "nyc").devices
+    assert "nyc-spine1__eth0" in exception_backend_a.get(exception_backend_a.device, "nyc-spine1").interfaces
+    assert exception_backend_a.get(exception_backend_a.interface, "nyc-spine1__eth0")
+
+    diff = exception_backend_a.diff_from(backend_a)
+    assert diff.has_diffs()
+
+    # Sync a second time
+    exception_backend_a.sync_from(backend_a)
+    assert "nyc-spine1" in exception_backend_a.get(exception_backend_a.site, "nyc").devices
+    assert "nyc-spine1__eth0" in exception_backend_a.get(exception_backend_a.device, "nyc-spine1").interfaces
+    assert exception_backend_a.get(exception_backend_a.interface, "nyc-spine1__eth0")
+
+
+def test_diffsync_sync_with_crud_no_delete_flag_on_models_not_skip_children(backend_a, exception_backend_a):
+    missing_device = backend_a.get(backend_a.device, "nyc-spine1")
+    backend_a.get(backend_a.site, "nyc").remove_child(missing_device)
+    backend_a.remove(missing_device)
+
+    assert "nyc-spine1" in exception_backend_a.get(exception_backend_a.site, "nyc").devices
+    assert "nyc-spine1__eth0" in exception_backend_a.get(exception_backend_a.device, "nyc-spine1").interfaces
+    assert exception_backend_a.get(exception_backend_a.interface, "nyc-spine1__eth0")
+
+    model1 = exception_backend_a.get(exception_backend_a.device, "nyc-spine1")
+    model1.model_flags |= DiffSyncModelFlags.CRUD_NO_DELETE
+
+    diff = exception_backend_a.diff_from(backend_a)
+    assert diff.has_diffs()
+
+    exception_backend_a.sync_from(backend_a)
+    assert "nyc-spine1" in exception_backend_a.get(exception_backend_a.site, "nyc").devices
+    assert "nyc-spine1__eth0" not in exception_backend_a.get(exception_backend_a.device, "nyc-spine1").interfaces
```

Comment:
It's a little surprising to me that `CRUD_NO_DELETE` by itself doesn't prevent deletion of child objects. I can sort of understand requiring to also set `SKIP_CHILDREN_ON_DELETE` but it would be good to document this really clearly.

Reply:
Doc update 
```
| CRUD_NO_DELETE | Do not delete and do not call delete() on the DiffSyncModel during sync(), the model and the changes will still be visible in the diff. This flag do not impact the children of this object, it's recommended to either set the flag on all objects individually or set the flag SKIP_CHILDREN_ON_DELETE | 0b1000000 |
```